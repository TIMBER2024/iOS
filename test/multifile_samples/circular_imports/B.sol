import "./A.sol";

/// #invariant true;
contract B {
  function a(A a) external returns (A) { return a; }
}
