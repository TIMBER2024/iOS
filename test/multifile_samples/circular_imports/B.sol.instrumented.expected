import "./A.sol";
import "./__scribble_ReentrancyUtils.sol";

/// #invariant true;
contract B is __scribble_ReentrancyUtils {
    event AssertionFailed(string message);

    function a(A a) external returns (A RET_0) {
        __scribble_out_of_contract = false;
        RET_0 = _original_B_a(a);
        __scribble_check_state_invariants();
        __scribble_out_of_contract = true;
    }

    function _original_B_a(A a) private returns (A) {
        return a;
    }

    /// Check only the current contract's state invariants
    function __scribble_B_check_state_invariants_internal() internal {
        unchecked {
            if (!(true)) {
                emit AssertionFailed("1: ");
                assert(false);
            }
        }
    }

    /// Check the state invariant for the current contract and all its bases
    function __scribble_check_state_invariants() virtual internal {
        __scribble_B_check_state_invariants_internal();
    }

    constructor() {
        __scribble_out_of_contract = false;
        __scribble_check_state_invariants();
        __scribble_out_of_contract = true;
    }
}